tests/e2e/volume_health_timestamp.go

```diff
@@ -0,0 +1,172 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fnodes "k8s.io/kubernetes/test/e2e/framework/node"
+	fpv "k8s.io/kubernetes/test/e2e/framework/pv"
+)
+
+var _ = ginkgo.Describe("Volume health timestamp check", func() {
+	f := framework.NewDefaultFramework("volume-health-timestamp-check")
+	var (
+		client            clientset.Interface
+		namespace         string
+		scParameters      map[string]string
+		storagePolicyName string
+		hostIP            string
+		pvclaim           *v1.PersistentVolumeClaim
+	)
+	ginkgo.BeforeEach(func() {
+		bootstrap()
+		client = f.ClientSet
+		namespace = getNamespaceToRunTests(f)
+		scParameters = make(map[string]string)
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		//datastoreURL := GetAndExpectStringEnvVar(envSharedDatastoreURL)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if guestCluster {
+			svcClient, svNamespace := getSvcClientAndNamespace()
+			setResourceQuota(svcClient, svNamespace, rqLimit)
+		}
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+
+	})
+
+	ginkgo.AfterEach(func() {
+		if supervisorCluster {
+			deleteResourceQuota(client, namespace)
+		}
+		if pvclaim != nil {
+			if hostIP != "" {
+				ginkgo.By("checking host status")
+				err := waitForHostToBeUp(hostIP)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+			err := fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+		if guestCluster {
+			svcClient, svNamespace := getSvcClientAndNamespace()
+			setResourceQuota(svcClient, svNamespace, defaultrqLimit)
+		}
+
+	})
+
+	/*
+		Test to verify health annotation status is accessible on the pvc .
+		(Combined test for TC1 and TC2)
+		Steps
+			1.	Create a Storage Class
+			2.	Create a PVC using above SC
+			3.	Wait for PVC to be in Bound phase
+			4.	Verify health annotation is added on the PVC is accessible
+			5.	Wait for the CNS health api to be called again (No changes made to PV/PVC, expecting it to be accessible)
+			6.	Delete PVC
+			7.	Verify PV entry is deleted from CNS
+			8.	Delete the SC
+	*/
+
+	ginkgo.It("[csi-supervisor] [csi-guest] Verify health timestamp annotation is added on the pvc", func() {
```

Comment:
this looks like a generic flow, why add a new test case and script can we combine this with any of the existing testcases

Reply:
In future, if we need to add scenarios around this timestamp feature then it will be easy for us to add scenarios hence created a new file.

Reply:
I still feel we should merge this test with existing test, we can create this new file if needed

Reply:
done, I have added the test in the existing file